ntoskrnl/io/iomgr/deviface.c

```diff
@@ -30,185 +31,421 @@ static PWCHAR BaseKeyString = L"\\Registry\\Machine\\System\\CurrentControlSet\\
 
 static
 NTSTATUS
-OpenRegistryHandlesFromSymbolicLink(IN PUNICODE_STRING SymbolicLinkName,
-                                    IN ACCESS_MASK DesiredAccess,
-                                    IN OPTIONAL PHANDLE GuidKey,
-                                    IN OPTIONAL PHANDLE DeviceKey,
-                                    IN OPTIONAL PHANDLE InstanceKey)
+IopBuildSymbolicLink(_In_ PUNICODE_STRING PrefixString,
+                     _In_ PUNICODE_STRING DeviceString,
+                     _In_ PUNICODE_STRING GuidString,
+                     _In_opt_ PUNICODE_STRING ReferenceString,
+                     _Out_ PUNICODE_STRING SymbolicLinkName)
 {
-    OBJECT_ATTRIBUTES ObjectAttributes;
-    UNICODE_STRING BaseKeyU;
-    UNICODE_STRING GuidString, SubKeyName, ReferenceString;
-    PWCHAR StartPosition, EndPosition;
-    HANDLE ClassesKey;
-    PHANDLE GuidKeyRealP, DeviceKeyRealP, InstanceKeyRealP;
-    HANDLE GuidKeyReal, DeviceKeyReal, InstanceKeyReal;
+    UNICODE_STRING SymbolicLink;
     NTSTATUS Status;
 
-    SubKeyName.Buffer = NULL;
+    /* Build up new symolic link */
+    SymbolicLink.Length = 0;
+    SymbolicLink.MaximumLength = PrefixString->Length + DeviceString->Length + sizeof(L"#") + GuidString->Length + ReferenceString->Length + sizeof(UNICODE_NULL);
+    SymbolicLink.Buffer = ExAllocatePoolWithTag(NonPagedPool, SymbolicLink.MaximumLength, TAG_IO);
```

Comment:
I think everything in this file is okay to use `PagedPool`

Reply:
Indeed. After changing the pool type to paged, nothing became worse at least. :slightly_smiling_face: